files/opt/upgrade.sh

```diff
@@ -10,25 +10,25 @@ if [[ $REPLY =~ ^[Yy]$|^$ ]]; then
     yarn set version classic
 
     echo "Downloading new Mastodon code..."
-    su - mastodon -c "cd /home/mastodon/live && git fetch --tags && git checkout $(git tag -l | grep '^v[0-9.]*$' | sort -V | tail -n 1)"
+    su - mastodon -c "cd /home/mastodon/live && git fetch --tags && git checkout $(su - mastodon -c 'cd /home/mastodon/live && git tag -l | grep '^v[0-9.]*$' | sort -V | tail -n 1')"
     RUBY_VERSION=$(cat /home/mastodon/live/.ruby-version)
 
     echo "Stopping Mastodon services..."
-    systemctl stop mastodon-web 
-    systemctl stop mastodon-streaming 
-    systemctl stop mastodon-sidekiq 
+    systemctl stop mastodon-web
+    systemctl stop mastodon-streaming
+    systemctl stop mastodon-sidekiq
 
     echo "Upgrading Ruby..."
-    su - mastodon -c "cd /home/mastodon/live && RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install $RUBY_VERSION && rbenv global $RUBY_VERSION"
+    su - mastodon -c "cd /home/mastodon/live && RUBY_CONFIGURE_OPTS=--with-jemalloc /home/mastodon/.rbenv/bin/rbenv install $RUBY_VERSION && /home/mastodon/.rbenv/bin/rbenv global $RUBY_VERSION"
```

Comment:
I wonder why the `PATH` is not correctly set here. `su -` (or `su -l`) should execute the shell rc files, and set up `rbenv` correctly

Reply:
I'm not really sure why it doesn't work either. My theory is, since this file is called by `setup.sh`, and is technically s subshell, this is creating a subshell within a subshell, and maybe that's causing some strange side-effect?

Reply:
Probably becasue the PATH are added at the end of the .bashrc default file, whose first action is to test if being run interactively:
```
# If not running interactively, don't do anything                   
case $- in                                                          
    *i*) ;;
      *) return;;
esac  
```

Reply:
Yup that was indeed the issue. Thanks for the insight!